File: DevPulse/DevPulse.Application/DTOs/Projects/UserListItemDto.cs
namespace DevPulse.Application.DTOs.Projects;

public class UserListItemDto
{
    public Guid Id { get; set; }

    public string Email { get; set; }

    public string FullName { get; set; }

    public bool IsAdmin { get; set; }

    public bool IsActive { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime? LastLoginAt { get; set; }

    public int ProjectCount { get; set; }
}

File: admin-panel/src/shared/types/UserListItemDto.ts
export interface UserListItemDto {
  id: string;
  email: string;
  fullName: string;
  isAdmin: boolean;
  isActive: boolean;
  createdAt: string;
  lastLoginAt?: string | null;
  projectCount: number;
}

export interface GetUsersParams {
  page: number;
  pageSize: number;
  search?: string;
  sortField?: string;
  sortDirection?: "asc" | "desc";
}

/** @deprecated Use GetUsersParams */
export type UsersParam = GetUsersParams;

/** @deprecated Use GetUsersParams */
export type UseUsersParams = GetUsersParams;

export interface CreateUserRequest {
  email: string;
  password: string;
  fullName: string;
}
